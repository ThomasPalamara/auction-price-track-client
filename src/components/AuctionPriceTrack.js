import React from 'react';
import RealmSelection from './RealmSelection';
import MainList from './MainList';
import DisplayStats from './DisplayStats';
import { Divider } from 'antd';

// TODO Ajouter la selection de language au store puis corrigÃ© celle en place et ajouter a display table

export default class AuctionPriceTrack extends React.Component {
  state = {
    realm : 'Ysondre',
    auctionHouse: null,
    loading: false,
  };
  handleRealmPicked = (realm) => {
    this.setState({ realm });
  }

  componentDidUpdate(prevProps, prevState) {
    if (this.state.realm && prevState.realm !== this.state.realm) {
      console.log('Fetching');
      this.setState({ loading: true });
      fetch(`/api/auctions/${this.state.realm}`)
        .then(res => res.json())
        .then(response => {
          this.setState({ loading: false });
          response.map((item) => {
            return item.auctions[0].prices.sort(function (a, b) { return (a.price > b.price) ? 1 : ((b.price > a.price) ? -1 : 0); });
          })
          return this.setState({ auctionHouse: response })
        }
      );
    }
  }

    render() {
    let displayPriceSection;
    if(!this.state.auctionHouse){
      displayPriceSection = <h3>Veuillez selectionner un serveur</h3>;
    }else{
      displayPriceSection = <DisplayStats auctionHouse={this.state.auctionHouse}/>
    }
    return (
      <div className="main">
        <RealmSelection loading={this.state.loading} handleRealmPicked={this.handleRealmPicked}/>
        <Divider/>
        <MainList/>
        <Divider/>
        {displayPriceSection}
      </div>
    );
  }
}
