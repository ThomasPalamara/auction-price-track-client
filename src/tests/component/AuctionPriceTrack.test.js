import React from 'react';
import { shallow } from "enzyme";
import toJSON from 'enzyme-to-json';
import AuctionPriceTrack from '../../components/AuctionPriceTrack';

test('Should render AuctionPriceTrack component correctly', () => {
    const wrapper = shallow(<AuctionPriceTrack/>);
    expect(toJSON(wrapper)).toMatchSnapshot();
    expect(wrapper.find('h3').length).toBe(1);
})

test('Should render DisplayStats component correctly when server set', () => {
    const wrapper = shallow(<AuctionPriceTrack/>);
    wrapper.setState({auctionHouse: {data:'some data'}})
    expect(toJSON(wrapper)).toMatchSnapshot();
    expect(wrapper.find('DisplayStats').length).toBe(1);
})