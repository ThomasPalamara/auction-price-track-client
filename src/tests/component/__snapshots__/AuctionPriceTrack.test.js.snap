// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render AuctionPriceTrack component correctly 1`] = `
<div
  className="main"
>
  <realmSelection
    handleRealmPicked={[Function]}
    loading={false}
  />
  <Divider />
  <Connect(MainList) />
  <Divider />
  <h3>
    Veuillez selectionner un serveur
  </h3>
</div>
`;

exports[`Should render DisplayStats component correctly when server set 1`] = `
<div
  className="main"
>
  <realmSelection
    handleRealmPicked={[Function]}
    loading={false}
  />
  <Divider />
  <Connect(MainList) />
  <Divider />
  <DisplayStats
    auctionHouse={
      Object {
        "data": "some data",
      }
    }
  />
</div>
`;
